src/MGroup.Constitutive.ConvectionDiffusion/ProblemConvectionDiffusion.cs

```diff
@@ -0,0 +1,287 @@
+		}
+
+		public void Reset()
+		{
+			ClearMatrices();
+		}
+
+		public void GetProblemDofTypes()
+		{
+			// Contents were commented out, function no longer used
+		}
+		#endregion
+
+		#region IImplicitIntegrationProvider Members
+
+		public void LinearCombinationOfMatricesIntoEffectiveMatrix(TransientAnalysisCoefficients coefficients)
+		{
+			IGlobalMatrix matrix = Diffusion;
+			matrix.AddIntoThis(Convection);
+			matrix.AddIntoThis(Production);
+			matrix.AxpyIntoThis(CapacityMatrix, coefficients.FirstOrderDerivativeCoefficient);
+			solver.LinearSystem.Matrix = matrix;
+			shouldRebuildDiffusionMatrixForZeroOrderDerivativeMatrixVectorProduct = true;
+		}
+
+		public void LinearCombinationOfMatricesIntoEffectiveMatrixNoOverwrite(TransientAnalysisCoefficients coefficients)
+		{
+			IGlobalMatrix matrix = Diffusion.Copy();
+			matrix.AddIntoThis(Convection);
+			matrix.AddIntoThis(Production);
+			matrix.AxpyIntoThis(CapacityMatrix, coefficients.FirstOrderDerivativeCoefficient);
+			solver.LinearSystem.Matrix = matrix;
+		}
+
+		public void ProcessRhs(TransientAnalysisCoefficients coefficients, IGlobalVector rhs)
+		{
+			// Method intentionally left empty.
+		}
+
+		public IGlobalVector GetSecondOrderDerivativeVectorFromBoundaryConditions(double time) => algebraicModel.CreateZeroVector();
+
+		public IGlobalVector GetFirstOrderDerivativeVectorFromBoundaryConditions(double time)
+		{
+			var boundaryConditions = model.EnumerateBoundaryConditions(model.EnumerateSubdomains().First().ID).ToArray();
+			foreach (var boundaryCondition in boundaryConditions.OfType<ITransientBoundaryConditionSet<IConvectionDiffusionDofType>>())
+			{
+				boundaryCondition.CurrentTime = time;
+			}
+
+			IGlobalVector velocities = algebraicModel.CreateZeroVector();
+			algebraicModel.AddToGlobalVector(
+			id =>
+			{
+				var boundaryConditions = model.EnumerateBoundaryConditions(id);
+				foreach (var boundaryCondition in boundaryConditions.OfType<ITransientBoundaryConditionSet<IConvectionDiffusionDofType>>().ToArray())
+				{
+					boundaryCondition.CurrentTime = time;
+				}
+
+				// TODO: Is this right?
+				return boundaryConditions
+					.SelectMany(x => x.EnumerateNodalBoundaryConditions())
+					.OfType<INodalCapacityBoundaryCondition>();
+			},
+			velocities);
+
+			// TODO: Is this right?
+			algebraicModel.AddToGlobalVector(boundaryConditions
+				.SelectMany(x => x.EnumerateDomainBoundaryConditions())
+				.OfType<IDomainCapacityBoundaryCondition>(),
+			velocities);
+
+			return velocities;
+		}
+
+		public IGlobalVector GetRhs(double time)
+		{
+			solver.LinearSystem.RhsVector.Clear();
+			AssignRhs();
+			IGlobalVector result = solver.LinearSystem.RhsVector.Copy();
+
+			return result;
+		}
+
+		public IGlobalVector SecondOrderDerivativeMatrixVectorProduct(IGlobalVector vector) => algebraicModel.CreateZeroVector();
+
+		public IGlobalVector FirstOrderDerivativeMatrixVectorProduct(IGlobalVector vector)
+		{// TODO: is the matrix right?
```

Comment:
Provide tests so that there are no open questions about the code functioning properly

Reply:
Tests are ready to be deployed on MSolve.Tests as soon as this pull request gets accepted. These are old comments that need to be removed, as per your suggestion above